common4j/src/main/com/microsoft/identity/common/java/telemetry/events/ApiStartEvent.java

```diff
@@ -0,0 +1,248 @@
+
+        if (parameters instanceof BrokerSilentTokenCommandParameters) {
+            final BrokerSilentTokenCommandParameters silentParameters = (BrokerSilentTokenCommandParameters) parameters;
+        }
+
+        return this;
+    }
+
+    public ApiStartEvent authority(@NonNull final String authority) {
+        put(Key.AUTHORITY, sanitizeUrlForTelemetry(authority));
+        return this;
+    }
+
+    public ApiStartEvent putAuthorityType(@NonNull final String authorityType) {
+        put(Key.AUTHORITY_TYPE, authorityType);
+        return this;
+    }
+
+    public ApiStartEvent putApiId(@NonNull final String apiId) {
+        put(Key.API_ID, apiId);
+        return this;
+    }
+
+    public ApiStartEvent putValidationStatus(@NonNull final String validationStatus) {
+        put(Key.AUTHORITY_VALIDATION_STATUS, validationStatus);
+        return this;
+    }
+
+    public ApiStartEvent putLoginHint(@NonNull final String loginHint) {
+        final String methodTag = TAG + ":putLoginHint";
+        try {
+            put(Key.LOGIN_HINT, StringUtil.createHash(loginHint));
+        } catch (final NoSuchAlgorithmException exception) {
+            Logger.warn(methodTag, exception.getMessage());
+        }
+
+        return this;
+    }
+
+    public ApiStartEvent putExtendedExpiresOnSetting(@NonNull final String extendedExpiresOnSetting) {
+        put(Key.EXTENDED_EXPIRES_ON_SETTING, extendedExpiresOnSetting);
+        return this;
+    }
+
+    /**
+     * Convenience method to sanitize the url for telemetry.
+     *
+     * @param url the {@link URL} to sanitize.
+     * @return the sanitized URL.
+     */
+    private static String sanitizeUrlForTelemetry(@NonNull final String url) {
+        URL urlToSanitize = null;
+        try {
+            urlToSanitize = new URL(url);
+        } catch (MalformedURLException e1) {
+            Logger.errorPII(
+                    TAG,
+                    "Url is invalid",
+                    e1
+            );
+        }
+
+        return urlToSanitize == null ? null : sanitizeUrlForTelemetry(urlToSanitize);
+    }
+
+    /**
+     * Sanitizes {@link URL} of tenant identifiers. B2C authorities are treated as null.
+     *
+     * @param url the URL to sanitize.
+     * @return the sanitized URL.
+     */
+    private static String sanitizeUrlForTelemetry(@NonNull final URL url) {
+        final String authority = url.getAuthority();
+        final String[] splitArray = url.getPath().split("/");
+
+        final StringBuilder logPath = new StringBuilder();
+        logPath.append(url.getProtocol())
+                .append("://")
+                .append(authority)
+                .append('/');
+
+        // we do not want to send tenant information
+        // index 0 is blank
+        // index 1 is tenant
+        for (int i = 2; i < splitArray.length; i++) {
+            logPath.append(splitArray[i]);
+            logPath.append('/');
+        }
+
+        return logPath.toString();
```

Comment:
I'm wondering if instead of doing all this, we can just grab the host and send that